[Bug]: Unable to parse rolldown.config.ts that are not in the working directory

### Reproduction link or steps

build/rolldown.config.ts:
```ts
import { defineConfig } from 'rolldown'

export default defineConfig({
  input: 'main.ts',
})
```

run `pnpm rolldown -c build/rolldown.config.ts`

### What is expected?

Can build main.ts

### What is actually happening?

output
```
Error: Error happened while loading config.
    at loadConfig (file:///Users/asnow/project/study/rolldown/node_modules/.pnpm/rolldown@1.0.0-beta.1/node_modules/rolldown/dist/esm/cli.mjs:352:9)
    at async bundleWithConfig (file:///Users/asnow/project/study/rolldown/node_modules/.pnpm/rolldown@1.0.0-beta.1/node_modules/rolldown/dist/esm/cli.mjs:359:17)
    at async main (file:///Users/asnow/project/study/rolldown/node_modules/.pnpm/rolldown@1.0.0-beta.1/node_modules/rolldown/dist/esm/cli.mjs:2004:3) {
  [cause]: Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/asnow/project/study/rolldown/rolldown.config.CfaagXO5.js' imported from /Users/asnow/project/study/rolldown/node_modules/.pnpm/rolldown@1.0.0-beta.1/node_modules/rolldown/dist/esm/cli.mjs
      at finalizeResolution (node:internal/modules/esm/resolve:275:11)
      at moduleResolve (node:internal/modules/esm/resolve:932:10)
      at defaultResolve (node:internal/modules/esm/resolve:1056:11)
      at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:654:12)
      at #cachedDefaultResolve (node:internal/modules/esm/loader:603:25)
      at ModuleLoader.resolve (node:internal/modules/esm/loader:586:38)
      at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:242:38)
      at onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:546:36)
      at TracingChannel.tracePromise (node:diagnostics_channel:344:14)
      at ModuleLoader.import (node:internal/modules/esm/loader:545:21) {
    code: 'ERR_MODULE_NOT_FOUND',
    url: 'file:///Users/asnow/project/study/rolldown/rolldown.config.CfaagXO5.js'
  }
}
```


### System Info

```Shell
System:
    OS: macOS 14.5
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Memory: 277.67 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.12.0 - ~/Library/pnpm/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 10.9.0 - ~/Library/pnpm/npm
    pnpm: 9.15.1 - ~/Library/pnpm/pnpm
    bun: 1.1.20 - ~/.bun/bin/bun
  Browsers:
    Edge: 131.0.2903.112
    Safari: 17.5
```

### Any additional comments?

_No response_

--- a/packages/rolldown/src/cli/load-config.ts
+++ b/packages/rolldown/src/cli/load-config.ts
@@ -52,16 +52,17 @@ async function bundleTsConfig(configFile: string): Promise<string> {
       },
     ],
   })
+  const outputDir = path.dirname(configFile)
   const result = await bundle.write({
-    dir: path.dirname(configFile),
+    dir: outputDir,
     format: 'esm',
     sourcemap: 'inline',
     entryFileNames: 'rolldown.config.[hash].js',
   })
-
-  return result.output.find(
+  const basename = result.output.find(
     (chunk): chunk is OutputChunk => chunk.type === 'chunk' && chunk.isEntry,
   )!.fileName
+  return path.resolve(outputDir, basename)
 }
 
 const SUPPORTED_JS_CONFIG_FORMATS = ['.js', '.mjs', '.cjs']

--- a/packages/rolldown/src/cli/load-config.ts
+++ b/packages/rolldown/src/cli/load-config.ts
@@ -8,7 +8,7 @@ import { OutputChunk } from '../types/rolldown-output'
 export async function loadTsConfig(configFile: string): Promise<ConfigExport> {
   const file = await bundleTsConfig(configFile)
   try {
-    return (await import(pathToFileURL(file).href)).default
+    return (await import(file)).default
   } finally {
     fs.unlink(file, () => {}) // Ignore errors
   }
@@ -18,7 +18,6 @@ async function bundleTsConfig(configFile: string): Promise<string> {
   const dirnameVarName = 'injected_original_dirname'
   const filenameVarName = 'injected_original_filename'
   const importMetaUrlVarName = 'injected_original_import_meta_url'
-
   const bundle = await rolldown({
     input: configFile,
     platform: 'node',
@@ -59,10 +58,10 @@ async function bundleTsConfig(configFile: string): Promise<string> {
     sourcemap: 'inline',
     entryFileNames: 'rolldown.config.[hash].js',
   })
-  const basename = result.output.find(
+  const fileName = result.output.find(
     (chunk): chunk is OutputChunk => chunk.type === 'chunk' && chunk.isEntry,
   )!.fileName
-  return path.resolve(outputDir, basename)
+  return path.join(outputDir, fileName)
 }
 
 const SUPPORTED_JS_CONFIG_FORMATS = ['.js', '.mjs', '.cjs']
@@ -74,6 +73,7 @@ const SUPPORTED_CONFIG_FORMATS = [
 
 export async function loadConfig(configPath: string): Promise<ConfigExport> {
   const ext = path.extname(configPath)
+  const rawConfigPath = path.resolve(configPath)
 
   try {
     if (
@@ -81,9 +81,9 @@ export async function loadConfig(configPath: string): Promise<ConfigExport> {
       (process.env.NODE_OPTIONS?.includes('--import=tsx') &&
         SUPPORTED_TS_CONFIG_FORMATS.includes(ext))
     ) {
-      return (await import(pathToFileURL(configPath).href)).default
+      return (await import(rawConfigPath)).default
     } else if (SUPPORTED_TS_CONFIG_FORMATS.includes(ext)) {
-      return await loadTsConfig(configPath)
+      return await loadTsConfig(rawConfigPath)
     } else {
       throw new Error(
         `Unsupported config format. Expected: \`${SUPPORTED_CONFIG_FORMATS.join(',')}\` but got \`${ext}\``,

--- a/packages/rolldown/src/cli/load-config.ts
+++ b/packages/rolldown/src/cli/load-config.ts
@@ -8,7 +8,7 @@ import { OutputChunk } from '../types/rolldown-output'
 export async function loadTsConfig(configFile: string): Promise<ConfigExport> {
   const file = await bundleTsConfig(configFile)
   try {
-    return (await import(file)).default
+    return (await import(pathToFileURL(file).href)).default
   } finally {
     fs.unlink(file, () => {}) // Ignore errors
   }
@@ -73,7 +73,6 @@ const SUPPORTED_CONFIG_FORMATS = [
 
 export async function loadConfig(configPath: string): Promise<ConfigExport> {
   const ext = path.extname(configPath)
-  const rawConfigPath = path.resolve(configPath)
 
   try {
     if (
@@ -81,8 +80,9 @@ export async function loadConfig(configPath: string): Promise<ConfigExport> {
       (process.env.NODE_OPTIONS?.includes('--import=tsx') &&
         SUPPORTED_TS_CONFIG_FORMATS.includes(ext))
     ) {
-      return (await import(rawConfigPath)).default
+      return (await import(pathToFileURL(configPath).href)).default
     } else if (SUPPORTED_TS_CONFIG_FORMATS.includes(ext)) {
+      const rawConfigPath = path.resolve(configPath)
       return await loadTsConfig(rawConfigPath)
     } else {
       throw new Error(

--- a/packages/rolldown/tests/cli/cli-e2e.test.ts
+++ b/packages/rolldown/tests/cli/cli-e2e.test.ts
@@ -148,6 +148,13 @@ describe('config', () => {
       expect(cleanStdout(status.stdout)).toMatchSnapshot()
     })
 
+    it('should allow loading ts config from non-working dir', async () => {
+      const cwd = cliFixturesDir()
+      const status = await $({ cwd })`rolldown -c ./ext-ts/rolldown.config.ts`
+      expect(status.exitCode).toBe(0)
+      expect(cleanStdout(status.stdout)).toMatchSnapshot()
+    })
+
     it('should allow multiply options', async () => {
       const cwd = cliFixturesDir('config-multiply-options')
       const status = await $({

--- a/packages/rolldown/tests/cli/fixtures/ext-ts/rolldown.config.ts
+++ b/packages/rolldown/tests/cli/fixtures/ext-ts/rolldown.config.ts
@@ -1,3 +1,4 @@
 export default {
   input: 'index.ts',
+  cwd: import.meta.dirname,
 }

